modular_zubbers/code/game/objects/devices/scanners/health_analyzer.dm

```diff
@@ -0,0 +1,483 @@
+GLOBAL_LIST_INIT(analyzerthemes, list(
+	"default",
+	"hackerman",
+	"ntos_rusty",
+	"ntos_healthy",
+))
+
+#define MAX_HEALTH_ANALYZER_UPDATE_RANGE 3
+
+/obj/item/healthanalyzer/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_ITEM_DROPPED, PROC_REF(on_drop))
+
+/obj/item/healthanalyzer/Destroy()
+	. = ..()
+	UnregisterSignal(src, COMSIG_ITEM_DROPPED, PROC_REF(on_drop))
+
+/obj/item/healthanalyzer/process(seconds_per_tick)
+	if(get_turf(src) != get_turf(current_user) || get_dist(get_turf(current_user), get_turf(patient)) > MAX_HEALTH_ANALYZER_UPDATE_RANGE || patient == current_user)
+		STOP_PROCESSING(SSobj, src)
+		patient = null
+		current_user = null
+		return
+	update_static_data(current_user)
```

Comment:
Move the static stuff to ui_data if you intend on having it update every process tick.

That or tie it to the signal for health updating

Reply:
if I change the static data, ui will be updating even if its not suposed to. Having the UI open will make you monitor the patient from whatever place you are. Same with camera monitor, if you have window open. You have acces to the camera no matter where you are.

Reply:
> if I change the static data, ui will be updating even if its not suposed to. Having the UI open will make you monitor the patient from whatever place you are. Same with camera monitor, if you have window open. You have acces to the camera no matter where you are.

Regular ui_data is called in every process tick too... what you're talking about is ui_status

Reply:
Basically, the checks should be part of ui_status and not inside process, and all the ui static data should just be ui_data since you seem to update it every tick anyhow

Reply:
Processing ticks are short, and they always are killed when user is not using the UI. I prefer the static data, as it prevents the data being updated even if you are 9999 tiles away from user.